File: test-suite/src/test/java/io/micronaut/docs/context/events/application/SampleEventListener.java
package io.micronaut.docs.context.events.application;

// tag::imports[]
import io.micronaut.context.event.ApplicationEventListener;
import io.micronaut.docs.context.events.SampleEvent;
// end::imports[]
import javax.inject.Singleton;

// tag::class[]
@Singleton
public class SampleEventListener implements ApplicationEventListener<SampleEvent> {
    private int invocationCounter = 0;

    @Override
    public void onApplicationEvent(SampleEvent event) {
        invocationCounter++;
    }

    public int getInvocationCounter() {
        return invocationCounter;
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/context/events/application/SampleEventListener.kt
package io.micronaut.docs.context.events.application

// tag::imports[]
import io.micronaut.context.event.ApplicationEventListener
import io.micronaut.docs.context.events.SampleEvent
// end::imports[]
import javax.inject.Singleton

// tag::class[]
@Singleton
class SampleEventListener : ApplicationEventListener<SampleEvent> {
    var invocationCounter = 0

    override fun onApplicationEvent(event: SampleEvent) {
        invocationCounter++
    }
}
// end::class[]